### PR TITLE
Add Formatting job

### PR DESCRIPTION
**Pull request by adiaaida**
This change adds the Windows and Ubuntu formatting jobs to the lab, and
makes them requestable. This also enables them to be run on push. The
jobs do the following:

On Windows:

- Build the Checked/Debug/Release builds as usenmakemakefiles, which
  is a configureonly option (so won't do a full build)
- Download the bootstrap script for jitutils
- Download the correct version of dotnet cli
- Run the bootstrap script
- Run jit-format for Windows_NT Checked/Debug/Release
  dll/standalone/crossgen configurations

On Ubuntu:

- Build release, checked, debug builds of coreclr in configureonly mode.
  This will only run cmake and generate the compile_commands.json
- Install the correct version of dotnet cli
- Install the bootstrap script for jitutils
- Run the bootstrap script
- Run jit format over all configurations for Ubuntu (release, checked,
  debug x dll, standalone, crossgen)

### CODE REVIEW

**Comment by adiaaida** [comment]
test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -1512,6 +1543,37 @@ combinedScenarios.each { scenario ->
                                         buildCommands += "build.cmd ${lowerConfiguration} ${arch} skiptests"
                                         buildCommands += "set __TestIntermediateDir=int&&build-test.cmd ${lowerConfiguration} ${arch}"
                                     }
+                                    else if (scenario == 'formatting') {
+                                        // Build with NMake Makefiles for all configurations
+                                        buildCommands += "set __TestIntermediateDir=int&&build.cmd Debug ${arch} usenmakemakefiles"
```
Comment (jashook) [inline netci.groovy]: This logic is significantly harder to test embedded into the groovy script like this. Can you hoist this out into a cross platform script and just call that?

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@mmitche @RussKeldorph @jashook PTAL
/cc: @dotnet/jit-contrib

File: netci.groovy [hunk 2]
```diff
@@ -543,6 +552,15 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
                 }
                 break
             }
+
+            if (scenario == 'formatting') {
+                assert configuration == 'Checked'
+                if (os == 'Windows_NT' || os == 'Ubuntu') {
+                    Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} Formatting Build & Test", "(?i).*test\\W+formatting.*")
```
Comment (jashook) [inline netci.groovy]: Would you mind adding this to our PR trigger documentation?

File: tests/scripts/format.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
+import urllib
```
Comment (jashook) [inline tests/scripts/format.py]: Can you add the same header as the rest of the files in this directory?

File: tests/scripts/format.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    print(output)
+    print(error)
+
+    # Run jit-format
+    returncode = 0
+    os.environ["PATH"] += os.pathsep + os.path.join(coreclr, "jitutils", "bin")
+
+    for build in ["Checked", "Debug", "Release"]:
+        for project in ["dll", "standalone", "crossgen"]:
+            proc = subprocess.Popen(" ".join(["jit-format", "-a", arch, "-b", build, "-o", platform,
```
Comment (jashook) [inline tests/scripts/format.py]: Popen takes a list not a string to run

File: netci.groovy (hunk 2 above)
Comment (adiaaida) [inline netci.groovy]: Where is this?

File: tests/scripts/format.py (hunk 3 above)
Comment (adiaaida) [inline tests/scripts/format.py]: done.

File: netci.groovy (hunk 2 above)
Comment (adiaaida) [inline netci.groovy]: Ah, I think I figured out what you meant.

**Comment by jashook** [comment]
lgtm

**Comment by adiaaida** [comment]
@mmitche @RussKeldorph ping

**Comment by mmitche** [comment]
LGTM

**Comment by adiaaida** [comment]
@RussKeldorph PTAL

**Comment by RussKeldorph** [comment]
@adiaaida Sorry, I thought I responded to this on Friday, but I guess it didn't go through...can I blame my mobile? :)

LGTM

**Pull request merged**